### PR TITLE
Alerting: Fix preview getting the correct queries from the form

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fixes the eval request being wrong when the user press shift-enter in the alert query in the alert rule form.

We [added](https://github.com/grafana/grafana/pull/68619/files#diff-6c9d7793ae5d088b056f743df107749d54dd7768b4744d97d1131c2000d650cdR136-R142) this `setValue('queries', updatedQueries, { shouldValidate: false });` in `onChangeQueries` ,  in order to fix an stale preview result. 

In this method, the `updatedQueries` parameter does not include the `expressions`. Consequently, we were assigning an incorrect list of queries in the form, where queries should represent the complete list of data queries along with expressions.

I tested locally and everything works as expected, but I would like to verify with the reviewers that i don't miss any use case.

**Who is this feature for?**

All users.

**Special notes for your reviewer:**

**Before the change:**

https://github.com/grafana/grafana/assets/33540275/6a01c52c-eb29-4de2-84fc-2ada43f87fac

**After the change:**

https://github.com/grafana/grafana/assets/33540275/37a53204-6815-44f1-aa41-d9a5f0237a79

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx [hunk 1]
```diff
@@ -99,10 +100,9 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
         // Grafana Managed rules and recording rules do
         return;
       }
-
-      runQueries(getValues('queries'), condition || (getValues('condition') ?? ''));
+      runQueries(queriesInForm, condition || (getValues('condition') ?? ''));
```
[the remark this answers is not in the document]
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:103]: 💯 agree

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx [hunk 2]
```diff
@@ -186,8 +185,9 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
       // Invocation cycle => onChange -> dispatch(setDataQueries) -> onRunQueries -> setDataQueries Reducer
       // As a workaround we update form values as soon as possible to avoid stale state
       // This way we can access up to date queries in runQueriesPreview without waiting for re-render
-      setValue('queries', updatedQueries, { shouldValidate: false });
-
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:190]: `updatedQueries` do not contain expressions.

**Review by gillesdemey — APPROVED** [review]
Works like a charm, nice investigation! 🕵️ :shipit:

**Review by konrad147 — APPROVED** [review]
LGTM!

**Pull request merged**